src/shop-cart-item.js

```diff
@@ -219,11 +222,15 @@ class ShopCartItem extends Element {
 
   _setCartItem(quantity) {
     this.dispatchEvent(new CustomEvent('set-cart-item', {
-      bubbles: true, composed: true, detail: {
+      bubbles: true, composed: true, detail: { quantity }}));
+    store.dispatch({
+      type: 'set-cart-item',
```

Comment:
Ideally actions are statements of what happened, not requests. `cart-item-set` would be a suitable alternative.